client/src/api/datasets.ts

```diff
@@ -48,22 +48,22 @@ export async function purgeDataset(datasetId: string) {
     return data;
 }
 
-type CopyDatasetParamsType = GalaxyApiPaths["/api/histories/{history_id}/contents/{type}s"]["post"]["parameters"];
-type CopyDatasetBodyType = components["schemas"]["CreateHistoryContentPayload"];
+type CopyContentParamsType = GalaxyApiPaths["/api/histories/{history_id}/contents/{type}s"]["post"]["parameters"];
+type CopyContentBodyType = components["schemas"]["CreateHistoryContentPayload"];
 
-export async function copyDataset(
-    datasetId: CopyDatasetBodyType["content"],
-    historyId: CopyDatasetParamsType["path"]["history_id"],
-    type: CopyDatasetParamsType["path"]["type"] = "dataset",
-    source: CopyDatasetBodyType["source"] = "hda"
+export async function copyContent(
+    contentId: CopyContentBodyType["content"],
+    historyId: CopyContentParamsType["path"]["history_id"],
+    type: CopyContentParamsType["path"]["type"] = "dataset"
```

Comment:
This is ambigous, I would pass in `src` (at least ldda is another valid dataset type) and then figure out which type it is, or pass both if you want to support dce's as well.